### PR TITLE
tool: dump tool supports index

### PR DESCRIPTION
**Pull request by xiang90**

### CODE REVIEW

File: tools/etcd-dump-logs/main.go [hunk 1]
```diff
@@ -32,35 +32,45 @@ import (
...
 	var (
+		walsnap  walpb.Snapshot
 		snapshot *raftpb.Snapshot
 		err      error
 	)
 
-	if *snapfile == "" {
-		ss := snap.New(snapDir(*from))
-		snapshot, err = ss.Load()
+	if *index != 0 {
```
Comment (yichengq) [inline tools/etcd-dump-logs/main.go]: set `*index != 0` a variable, e.g., startFromIndex ?

File: tools/etcd-dump-logs/main.go [hunk 2]
```diff
@@ -32,35 +32,45 @@ import (
...
+		fmt.Println("start dumping log entries from index.", *index)
```
Comment (yichengq) [inline tools/etcd-dump-logs/main.go]: full stop should be after *index..

File: tools/etcd-dump-logs/main.go (hunk 1 above)
Comment (xiang90) [inline tools/etcd-dump-logs/main.go]: you do not like the deference symbol?

File: tools/etcd-dump-logs/main.go (hunk 2 above)
Comment (xiang90) [inline tools/etcd-dump-logs/main.go]: right. will fix.

File: tools/etcd-dump-logs/main.go (hunk 1 above)
Comment (yichengq) [inline tools/etcd-dump-logs/main.go]: the `*index != 0` is used again later, and it is clear to make it a bool var

**Comment by xiang90** [comment]
@yichengq refresh

File: tools/etcd-dump-logs/main.go (hunk 1 above)
Comment (xiang90) [inline tools/etcd-dump-logs/main.go]: ok

**Comment by yichengq** [comment]
lgtm

**Pull request merged**